src/aiida/manage/configuration/profile.py

```diff
@@ -29,41 +31,44 @@
 class Profile:
     """Class that models a profile as it is stored in the configuration file of an AiiDA instance."""
 
-    KEY_UUID = 'PROFILE_UUID'
-    KEY_DEFAULT_USER_EMAIL = 'default_user_email'
-    KEY_STORAGE = 'storage'
-    KEY_PROCESS = 'process_control'
-    KEY_STORAGE_BACKEND = 'backend'
-    KEY_STORAGE_CONFIG = 'config'
-    KEY_PROCESS_BACKEND = 'backend'
-    KEY_PROCESS_CONFIG = 'config'
-    KEY_OPTIONS = 'options'
-    KEY_TEST_PROFILE = 'test_profile'
+    KEY_UUID: str = 'PROFILE_UUID'
+    KEY_DEFAULT_USER_EMAIL: str = 'default_user_email'
+    KEY_STORAGE: str = 'storage'
+    KEY_PROCESS: str = 'process_control'
+    KEY_STORAGE_BACKEND: str = 'backend'
+    KEY_STORAGE_CONFIG: str = 'config'
+    KEY_PROCESS_BACKEND: str = 'backend'
+    KEY_PROCESS_CONFIG: str = 'config'
+    KEY_OPTIONS: str = 'options'
+    KEY_TEST_PROFILE: str = 'test_profile'
 
     # keys that are expected to be in the parsed configuration
-    REQUIRED_KEYS = (
+    REQUIRED_KEYS: tuple[str, str] = (
         KEY_STORAGE,
         KEY_PROCESS,
     )
 
-    def __init__(self, name: str, config: Mapping[str, Any], validate=True):
+    def __init__(
+        self, name: str, config: abc.Mapping[str, Any], config_folder: pathlib.Path | None = None, validate: bool = True
+    ):
         """Load a profile with the profile configuration."""
-        if not isinstance(config, collections.abc.Mapping):
-            raise TypeError(f'config should be a mapping but is {type(config)}')
+        _ = type_check(config, abc.Mapping)
         if validate and not set(config.keys()).issuperset(self.REQUIRED_KEYS):
             raise exceptions.ConfigurationError(
                 f'profile {name!r} configuration does not contain all required keys: {self.REQUIRED_KEYS}'
             )
 
-        self._name = name
-        self._attributes: Dict[str, Any] = deepcopy(config)
+        self._name: str = name
+        self._attributes: dict[str, Any] = cast(dict[str, Any], deepcopy(config))
```

Comment:
Why is this cast necessary? 

I makes this really hard to read. If mypy complains, I'd just add `type: ignore` here.

Reply:
Okay, removed.